### PR TITLE
stream: Relocate to cilium/stream

### PR DESCRIPTION
**Pull request by joamaki**
The stream package now lives in https://github.com/cilium/stream.
Remove pkg/stream and refactor use-sites to use cilium/stream instead.

### CODE REVIEW

File: go.mod
Comment (giorio94) [inline go.mod]: The patch version should be always set to 0 AFAIR, to avoid forcing every developer to align with that exact patch version.

File: daemon/cmd/local_node_sync_test.go
Comment (giorio94) [inline daemon/cmd/local_node_sync_test.go]: It would be probably great to have a [linter](https://github.com/daixiang0/gci) enforcing that the `github.com/cilium/*` imports form a separate block, otherwise they'll soon start to get coalescenced with the ones above.

**Review by giorio94 — CHANGES_REQUESTED** [review]
Overall looks good to me. One inline comment about the `go.mod` version having been likely unintentionally bumped in go.mod, and another about the possibility of enforcing imports ordering through a linter, which could be also addressed separately. The `go.mod` file picked also a conflict.

**Review by tommyp1ckles — APPROVED** [review]

File: daemon/cmd/local_node_sync_test.go
Comment (bimmlerd) [reply daemon/cmd/local_node_sync_test.go]: Hm. I would've expected `goimports` to flag this, but I'm not sure how we configured it. In any case, do we actually want a third block?

File: go.mod
Comment (bimmlerd) [reply go.mod]: I think this is lower bound by the version from all go.mods of the dependencies, hence from https://github.com/cilium/stream/blob/main/go.mod in this case. Would need to change that, first.

**Review by bimmlerd — APPROVED** [review]
CODEOWNERS LGTM

File: daemon/cmd/local_node_sync_test.go
Comment (giorio94) [reply daemon/cmd/local_node_sync_test.go]: AFAIR `goimports` is more opinionated and cannot be configured (except for the local-prefixes which form the third block) -- I had tested it, and it would not flag this in either way -- while `gci` allows you to [configure the specific format for the imports](https://golangci-lint.run/usage/linters/#gci).

> In any case, do we actually want a third block?

I'm not sure honestly, although generally speaking it seems reasonable to me. Regardless, it is probably a good idea to pick a format and strictly enforce it, to remove this ambiguity (although totally unrelated from this PR).

**Review by dylandreimerink — APPROVED** [review]
I see nothing that others have not already mentioned, approving, assuming other discussions will resolve

File: go.mod
Comment (joestringer) [reply go.mod]: :heavy_plus_sign: Please set this to 0. There's some context here and in linked issues: https://github.com/cilium/cilium/pull/28686 .

**Review by youngnick — APPROVED** [review]

File: go.mod
Comment (joamaki) [reply go.mod]: Thanks for catching this! I've fixed this in cilium/stream, bumped deps and fixed the version here.

File: daemon/cmd/local_node_sync_test.go
Comment (joamaki) [reply daemon/cmd/local_node_sync_test.go]: Yeah this is a bit annoying. I did a "sed s,pkg/stream,cilium/stream" and then goimports and that's how it wanted to format these. Fixed it with:

```
find . -name "*.go" -exec sed -i ':a;$!{N;/.*\n{/ba};s,\n\t"github.com/cilium/stream",\t"github.com/cilium/stream",g;P;D' {} \;
goimports -local github.com/cilium/cilium -w .
```

File: go.mod
Comment (giorio94) [reply go.mod]: Thanks!

**Review by giorio94 — APPROVED** [review]
Thanks!

**Review by squeed — APPROVED** [review]

**Review by rolinh — APPROVED** [review]
Vendor changes lgtm

**Pull request merged**
